pkg/vm/monitor.go

```diff
@@ -118,9 +119,20 @@ func (m *Module) monitor(ctx context.Context) error {
 }
 
 func (m *Module) monitorID(ctx context.Context, running map[string]Process, id string) error {
+	stub := stubs.NewProvisionStub(m.client)
 	log := log.With().Str("id", id).Logger()
 
-	if _, ok := running[id]; ok {
+	if ps, ok := running[id]; ok {
+		state, err := stub.GetWorkloadStatus(ctx, id)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get workload status for vm:%s ", id)
+		}
+		// state = nil when workload doesn't exist
+		if state == "" || state.IsAny(gridtypes.StateDeleted, gridtypes.StateError) {
```

Comment:
I don't like having an empty state as a `not found` indication since it's not explicit either return an extra bool, or return a completey new struct 
```go
type OptionStatus struct{
   Exists bool
   State ResultState
}
```